### PR TITLE
Report framework name in exception

### PR DESCRIPTION
**Pull request by davilimap**
## Bug

Fixes: The exception is only thrown when the shortFramework is null or empty
Regression: No  

## Fix

Details: Report the full framework name in case there is some other error happening  

## Testing/Validation

Tests Added: No  
Reason for not adding tests: Diagnostic improvement
Validation:  N/A

### CODE REVIEW

**Comment by nkolev92** [comment]
This is the issue tracking it. 
https://github.com/NuGet/Home/issues/8174 🙃 

@zivkan Thinks that there are other issues that he's now investigating.

**Comment by zivkan** [comment]
right. In the scenario of NuGet/Home#7717 it's going to give an error message saying that `_` is not a valid framework. This is better than saying that the empty string is not a valid framework, but it's still not enough information for customers to act on.

The problem is that the project system tells us during nomination that the target framework moniker is `_`, so it feels like a bug in the project system. But, I'm in the process of debugging 15.9 where the NuGetFrameworkException is not thrown, to understand what's being nominated there to understand if it's a regression in the project system or NuGet.

**Comment by davilimap** [comment]
Thanks for the heads up., sorry for lack of context. I noticed this error popping up a lot in CPS faults in 16.2 but was having a hard time finding out what was wrong.

**Comment by zivkan** [comment]
NuGet gets thousands of watson hits on unhandled NuGetFrameworkException, so I'd desperately love to fix the issue. It's just a huge coincidence that I started looking at this after lunch today :)

Tomorrow I'll have more information.

File: src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs [hunk 1]
```diff
@@ -166,7 +166,7 @@ public virtual string GetShortFolderName(IFrameworkNameProvider mappings)
                     throw new FrameworkException(string.Format(
                         CultureInfo.CurrentCulture,
                         Strings.InvalidFrameworkIdentifier,
-                        shortFramework));
+                        framework.Framework));
```
Comment (zivkan) [inline src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs:169]: Please add a test to `test/NuGet.Core.Tests/NuGet.Frameworks.Test/NuGetFrameworkTests.cs` to ensure the framework name is in the message

**Review by zivkan — CHANGES_REQUESTED** [review]
Let's add a test to reduce the risk of regression in the future, but otherwise we need this and is good to merge.

**Comment by nkolev92** [comment]
It's been a while since we've visited this PR :)
I think we decided to go a different direction with this issue. 

Closing this PR per that.

**Pull request closed without merging**